### ISSUE TITLE
Support HTTP Basic authentication in syslog/https.go

### ISSUE THREAD

**Issue by jochenehret** [cloudfoundry/loggregator-agent-release#29]
Hi Loggregator Team,

we are using syslog sinks that require HTTP Basic authentication. The credentials are specified in the URL which we bind to our apps:
```
https://<user>:<password>@<log stream app>
```
It seems that fasthttp does not support this out-of-the-box, see https://github.com/valyala/fasthttp/issues/79. So the syslog-agent would have to set the HTTP Authorization header in `egress/syslog/https.go`:
```
req.Header.Add("Authorization", "Basic <base64 encoded user:password>")
```
Can you please provide a fix?

Thanks and Best Regards,

  Jochen.

**Comment by MasslessParticle** [comment]
Hello,

We're having trouble reproducing this issue. We've created a couple of sample apps:
- golang, dumps headers: https://gist.github.com/MasslessParticle/6d8f9aa9f97c10c8b4ae150b9e13490d

- java, spring auth: https://github.com/loggregator/baajava

In both cases, when an agent-based drain is bound with
`cf drain <app name> https://user:password@<drain location>` we can see the Basic Auth header present in the logs. 

Can you try one or both of these apps and see if the experience differs?

**Comment by jochenehret** [comment]
I've created a drain service with the golang app above and then bound a simple go-hello app:
```
cf cups my-drain -l https://user1:password1@go-baa.<domain>
cf bind-service go-hello my-drain
```
HTTP Basic auth works as expected:
```
   2020-03-24T15:42:35.81+0100 [APP/PROC/WEB/0] OUT Authorization:  Basic dXNlcjE6cGFzc3dvcmQx
   (...)
   2020-03-24T15:42:35.81+0100 [APP/PROC/WEB/0] OUT Auth OK?  true
   2020-03-24T15:42:35.81+0100 [APP/PROC/WEB/0] OUT Username  user1
   2020-03-24T15:42:35.81+0100 [APP/PROC/WEB/0] OUT Password  password1
```
So there seems to be something wrong with our syslog drain app.

Thanks for your quick assistance!

**Comment by andrew-edgar** [comment]
This was a major issue for us.  The use of fasthttp as the client broke HTTPS integration to splunkcloud which through the new agents always returned a 403.

We had to roll back to the old syslog adapters.  I would go as far as saying the loggr-syslog-agents should not be used by anybody who may need to connect to endpoints with basic auth.  There should probably be a notice in cf-deployment on cfd13!

I provided information as to where in the fasthttp code the problem seems to occur to the loggregator team and I hope this is move to very high priority

**Comment by andrew-edgar** [comment]
I added a new story in fasthttp as with more debugging it looks like it fails when the URL has "path" elements.  It works fine if the URL has no path and is just a host.

**Comment by andrew-edgar** [comment]
fasthttp has just fixed this issue.  loggregator-agent-release will need to pickup 1.13.1 to get a version that works.

**Comment by pianohacker** [comment]
Fixed in release 6.0.1.
